client/src/Components/CreateNewRoom/CreateNewRoom.jsx

```diff
@@ -4,15 +4,20 @@ import axios from 'axios';
 
 import VoteButton from 'Components/VoteButton/index.js';
 import CreateNewRoomStyles from './CreateNewRoomStyles';
+import { useRoomContext } from 'Contexts/RoomContext';
 
 const CreateNewRoom = () => {
   const history = useHistory();
   const classes = CreateNewRoomStyles();
+  const { setErrorMsg } = useRoomContext();
 
   const getRoomNumber = async () => {
     try {
       const currentRoom = await axios.post('/api/session');
-      history.push(`/room/${currentRoom.data.room.id}/join`, { isAdmin: true });
+      if (currentRoom.data?.room?.id) history.push(`/room/${currentRoom.data.room.id}/join`, { isAdmin: true });
```

Comment:
Rozumiem dlaczego musiałeś tu ifa napisać ale generalnie to axios kiedy response code jest >= 400 rzuca błędem. I to właśnie wtedy (w sekcji catch) powinniśmy decydować o wy renderowaniu błędu.

Tu musiałeś to tak zrobić bo na back-endzie błąd jest zwracany ze kodem 200 (czyli że niby wszystko ok). Proponowałbym to poprawić w ramach tego brancha byś mógł już pozbyć się ifa. By to zrobić w pliku session.ts dwunastą linie trzeba podmienić na `    res.status(500).json(ex.message);`